### PR TITLE
feat: add composite type support for field resolvers

### PR DESCRIPTION
**Pull request by Noroth**
This PR adds support for using composite type in field resolvers.

## Summary by CodeRabbit

* **New Features**
  * Project.criticalDeadline — fetch upcoming milestones or project end dates within a configurable window.
  * Project.topPriorityItem — surface the highest-priority item (task, milestone, or project) with optional category filtering.

* **Bug Fixes**
  * Improved handling of list/multi-valued resolver arguments.
  * QueryProject now returns nil for not-found cases instead of an error.

* **Tests**
  * Greatly expanded coverage for field-resolver scenarios, lists, nested contexts, fragments, aliases and edge cases.

* **Chores**
  * Updated prerelease dependency entries.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [ ] I have followed the coding standards of the project.
- [ ] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/cosmo-docs](https://github.com/wundergraph/cosmo-docs).
- [ ] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Noroth** [comment]
@coderabbitai review

**Comment by Noroth** [comment]
Using temporary engine version to test CI

File: demo/pkg/subgraphs/projects/src/service/service.go [hunk 1]
```diff
@@ -38,6 +39,177 @@ type ProjectsService struct {
...
+
+	return response, nil
+}
+
+// ResolveProjectTopPriorityItem implements projects.ProjectsServiceServer.
+func (p *ProjectsService) ResolveProjectTopPriorityItem(ctx context.Context, req *service.ResolveProjectTopPriorityItemRequest) (*service.ResolveProjectTopPriorityItemResponse, error) {
+	logger := hclog.FromContext(ctx)
+
+	p.lock.RLock()
+	defer p.lock.RUnlock()
```
Comment (StarpTech) [inline demo/pkg/subgraphs/projects/src/service/service.go:127]: Why is a lock needed here?

File: protographic/src/sdl-to-proto-visitor.ts [hunk 2]
```diff
@@ -1294,8 +1294,8 @@ Example:
         messageName: contextMessageName,
         fields: fieldFilter.map((field) => ({
           fieldName: graphqlFieldToProtoField(field.name),
-          typeName: this.getProtoTypeFromGraphQL(field.type).typeName,
           fieldNumber: ++fieldNumber,
+          ...this.getProtoTypeFromGraphQL(field.type),
```
Comment (StarpTech) [inline protographic/src/sdl-to-proto-visitor.ts:1298]: Is this type safe or do we might include more than necessary in the future? If not a burden, I'd recommend being specific.

File: demo/pkg/subgraphs/projects/src/service/service.go (hunk 1 above)
Comment (Noroth) [reply demo/pkg/subgraphs/projects/src/service/service.go:127]: Oh just if any write operations are running at the same time. I just wanted to make sure that we don't provide incorrect data and have race conditions in the demo service.
